lib/dirac_twisted_clover.cpp

```diff
@@ -77,8 +79,10 @@ namespace quda {
   {
 
     if (in.TwistFlavor() == QUDA_TWIST_SINGLET) {
-      // k * D * in + (A + i*2*mu*kappa*gamma_5) *x
-      ApplyTwistedClover(out, in, *gauge, *clover, k, 2 * mu * kappa, x, parity, dagger, commDim, profile);
+      // k * D * in + (A + i*2*(mu+tm_rho)*kappa*gamma_5) *x
+      // tm_rho is a Hasenbusch mass preconditioning parameter applied just like a twisted mass
```

Comment:
in facts, where is the inverse applied? I can't see it in this file

Reply:
`DiracTwistedCloverPC::Dslash`

Reply:
more illustratively, the logic in `DiracTwistedCloverPC::M` should clarify what's going on

Reply:
ok